axelrod/strategies/axelrod_second.py

```diff
@@ -199,3 +199,60 @@ def strategy(self, opponent: Player) -> Action:
         else:
             # Play TFT
             return opponent.history[-1]
+
+
+class MoreGrofman(Player):
+    """
+    Submitted to Axelrod's second tournament by Bernard Grofman.
+
+    This strategy has 3 phases:
+    1. First it cooperates on the first two rounds
+    2. For rounds 3-7 inclusive, it plays the same as the opponent's last move
+    3. Thereafter, it applies the following logic
+
+      - If its own previous move was C and the opponent has defected less than
+        twice in the last 7 rounds, cooperate
+      - If its own previous move was C and the opponent has defected twice or
+        more in the last 7 rounds, defect
+      - If its own previous move was D and the opponent has defected only once
+        or not at all in the last 7 rounds, cooperate
+      - If its own previous move was D and the opponent has defected more than
+        once in the last 7 rounds, defect
+
+    Names:
+    - Grofman's strategy: [Axelrod1980b]_
+    """
+    name = "MoreGrofman"
+    classifier = {
+        'memory_depth': 7,
+        'stochastic': False,
+        'makes_use_of': set(),
+        'long_run_time': False,
+        'inspects_source': False,
+        'manipulates_source': False,
+        'manipulates_state': False
+    }
+
+    def __init__(self) -> None:
+        super().__init__()
+
+    def strategy(self, opponent: Player) -> Action:
+        # Cooperate on the first two moves
+        if not self.history or len(self.history) in [1]:
+            return C
+        # For rounds 3-7, play the opponent's last move
+        elif 2 <= len(self.history) <= 6:
+            return opponent.history[-1]
+        # Logic for the rest of the game
+        else:
+            opponent_defections_last_7_rounds = opponent.history[-8:-1].count(D) # (-7:) fails at 13, (-7:-1), (-8:), (-8:-1), (-6:), (-6:-1)
```

Comment:
This doesn't seem correct: why doesn't `opponent.history[-7:]` work?

```python
>>> ten_numbers = list(range(10))
>>> len(ten_numbers[-7:])  # 7 numbers
7
>>> len(ten_numbers[-7:-1])  # Only 6 numbers
 6
```

Reply:
I originally tried `[-7:]` but that resulted in what amounted to an off by one error for certain cases. This came to light when comparing the Python implementation with the Fortran implementation. I believe that the difference in indexes comes from the difference in Fortran indexing during do-loops/arrays (1 indexed and inclusive-inclusive) vs Python indexing into lists. The `[-8:-1]` indexing appears to match the behavior of the Fortran. I will remove the comment after because I was writing down potential off by one candidates to experiment with because I didn't really understand the Fortran indexing at the time.

Reply:
Ok can you adjust the docstring description then because they're not what you are implemented:

```
If its own previous move was C and the opponent has defected less than twice in the last 7 rounds
```

I think we should say something like the mouthful that is `the 7 rounds before the last one` ... ?